### PR TITLE
Transfoxl sequence classification

### LINKED ISSUE #7623: Implement PyTorch and/or TensorFlow sequence classification architectures for causal language models
# 🚀 Feature request

The architecture `GPT2ForSequenceClassification` was added in #7501 in PyTorch. It would be great to have it in TensorFlow (cf. issues #7622), but it would also be great to have it for other causal models: ~OpenAI GPT~, ~CTRL~ (PR opened @elk-cloner), ~TransfoXL~ (PR opened @spatil6)

Below is a list of items to follow to make sure the integration of such an architecture is complete:

- Implement `XXXForSequenceClassification` in `modeling_xxx.py` or `TFXXXForSequenceClassification` in `modeling_tf_xxx.py
- Test that architecture in `tests/test_modeling_xxx.py` or `tests/test_modeling_tf_xxx.py`
- Add that architecture to `__init__.py` and ` docs/source/model_doc/xxx.rst`.

Taking a look at the code changes in #7501 would be a good start.

A very good first issue to get acquainted with the library and its architectures!

### PR DESCRIPTION
**Pull request by spatil6**
This PR implements Sequence classification for Transformer XL model
TransfoxlForSequenceClassification uses the last token in order to do the classification, as other causal models (e.g. GPT-1,GPT-2) do.

Fixes #7623 (Partially)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [x] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [x] Did you write any new necessary tests?

## Who can review?
@LysandreJik

### CODE REVIEW

File: src/transformers/models/transfo_xl/modeling_transfo_xl.py [hunk 1]
```diff
@@ -1101,3 +1136,110 @@ def _resize_cutoffs(self, new_num_tokens, new_emb_size, new_embedding_shapes, la
...
+        logits = self.score(hidden_states)
+
+        if input_ids is not None:
+            batch_size, sequence_length = input_ids.shape[:2]
+        else:
+            batch_size, sequence_length = inputs_embeds.shape[:2]
+
+        '''assert (
+            self.config.pad_token_id is not None or batch_size == 1
+        ), "Cannot handle batch sizes > 1 if no padding token is defined."'''
```
Comment (LysandreJik) [inline src/transformers/models/transfo_xl/modeling_transfo_xl.py:1211]: Why is this commented out?

**Review by LysandreJik — COMMENTED** [review]
Things look good to me, except the assertion commented out. @patrickvonplaten @TevenLeScao, do you want to take a look?

**Comment by LysandreJik** [comment]
Same as GPT-2, this would benefit from also handling padding on the left; I'll work on this in another PR.

File: src/transformers/models/transfo_xl/modeling_transfo_xl.py (hunk 1 above)
Comment (spatil6) [reply src/transformers/models/transfo_xl/modeling_transfo_xl.py]: As there was no pad_token_id is defined in configuration file of this model, so I have commented out.
But now I have updated TransfoXLModelTester with pad_token_id.

Comment (spatil6) [reply src/transformers/models/transfo_xl/modeling_transfo_xl.py]: As there was no pad_token_id is defined in configuration file of this model, so I have commented out.
But now I have updated TransfoXLModelTester with pad_token_id.

**Comment by spatil6** [comment]
@LysandreJik , I'll raise new PR, there was some conflicts in it.
